src/vm/jithelpers.cpp

```diff
@@ -3028,6 +3028,106 @@ HCIMPLEND
 //
 //========================================================================
 
+//*************************************************************
+// Array allocation fast path for arrays of value type elements
+//
+
+HCIMPL2(Object*, JIT_NewArr1VC_MP_Portable, CORINFO_CLASS_HANDLE arrayTypeHnd_, INT_PTR size)
+{
+    FCALL_CONTRACT;
+
+    do
+    {
+        _ASSERTE(GCHeap::UseAllocationContexts());
+
+        // Do a conservative check here.  This is to avoid overflow while doing the calculations.  We don't
+        // have to worry about "large" objects, since the allocation quantum is never big enough for
+        // LARGE_OBJECT_SIZE.
+        //
+        // For Value Classes, this needs to be 2^16 - slack (2^32 / max component size), 
+        // The slack includes the size for the array header and round-up ; for alignment.  Use 256 for the
+        // slack value out of laziness.
+        SIZE_T componentCount = static_cast<SIZE_T>(size);
+        if (static_cast<SIZE_T>(size) >= static_cast<SIZE_T>(65535 - 256))
+        {
+            break;
+        }
+
+        // This is typically the only call in this function. Making the call early seems to be better, as it allows the compiler
+        // to use volatile registers for intermediate values. This reduces the number of push/pop instructions and eliminates
+        // some reshuffling of intermediate values into nonvolatile registers around the call.
+        Thread *thread = GetThread();
+
+        TypeHandle typeHandle(arrayTypeHnd_);
+        _ASSERTE(typeHandle.GetInternalCorElementType() == ELEMENT_TYPE_SZARRAY);
+        ArrayTypeDesc *arrayTypeDesc = typeHandle.AsArray();
+
+        _ASSERTE(arrayTypeDesc->GetArrayElementTypeHandle().GetSignatureCorElementType() != ELEMENT_TYPE_VOID); // this would need to throw
+
+        if (DATA_ALIGNMENT < sizeof(double))
```

Comment:
These checks are not needed here. The logic that selects the CORINFO_HELP_NEWARR_1_VC helper in `CEEInfo::getNewArrHelperStatic` takes care of them upfront.